exchanges/lbank/lbank_test.go

```diff
@@ -0,0 +1,341 @@
+package lbank
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/thrasher-/gocryptotrader/config"
+	"github.com/thrasher-/gocryptotrader/currency"
+	exchange "github.com/thrasher-/gocryptotrader/exchanges"
+	log "github.com/thrasher-/gocryptotrader/logger"
+)
+
+// Please supply your own keys here for due diligence testing
+const (
+	testAPIKey    = ""
+	testAPISecret = ""
+)
+
+var l Lbank
+var setupRan bool
+
+func TestSetDefaults(t *testing.T) {
+	l.SetDefaults()
+}
+
+func TestSetup(t *testing.T) {
+	if setupRan {
+		return
+	}
+	setupRan = true
+
+	t.Parallel()
+	l.SetDefaults()
+	l.APIKey = testAPIKey
+	l.APISecret = testAPISecret
+	cfg := config.GetConfig()
+	err := cfg.LoadConfig("../../testdata/configtest.json")
+	if err != nil {
+		t.Errorf("Test Failed - Lbank Setup() init error:, %v", err)
+	}
+	lbankConfig, err := cfg.GetExchangeConfig("Lbank")
+	lbankConfig.Websocket = true
+	if err != nil {
+		t.Errorf("Test Failed - Lbank Setup() init error: %v", err)
+	}
+
+	l.Setup(&lbankConfig)
+}
+
+func TestGetTicker(t *testing.T) {
+	TestSetup(t)
+	_, err := l.GetTicker("btc_usdt")
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetCurrencyPairs(t *testing.T) {
+	TestSetup(t)
+	_, err := l.GetCurrencyPairs()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetMarketDepths(t *testing.T) {
+	TestSetup(t)
+	_, err := l.GetMarketDepths("btc_usdt", "60", "1")
+	if err != nil {
+		t.Errorf("GetMarketDepth failed: %v", err)
+	}
+	a, _ := l.GetMarketDepths("btc_usdt", "60", "0")
+	if len(a.Asks) != 60 {
+		t.Errorf("length requested doesnt match the output")
+	}
+	_, err = l.GetMarketDepths("btc_usdt", "61", "0")
+	if err == nil {
+		t.Errorf("size is greater than the maximum allowed")
+	}
+}
+
+func TestGetTrades(t *testing.T) {
+	TestSetup(t)
+	_, err := l.GetTrades("btc_usdt", "600", fmt.Sprintf("%v", time.Now().Unix()))
+	if err != nil {
+		t.Error(err)
+	}
+	a, err := l.GetTrades("btc_usdt", "600", "0")
+	if len(a) != 600 && err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetKlines(t *testing.T) {
+	TestSetup(t)
+	_, err := l.GetKlines("btc_usdt", "600", "minute1", fmt.Sprintf("%v", time.Now().Unix()))
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	TestSetup(t)
+	p := currency.Pair{
+		Delimiter: "_",
+		Base:      currency.ETH,
+		Quote:     currency.BTC}
+
+	_, err := l.UpdateOrderbook(p.Lower(), "spot")
+	if err != nil {
+		t.Errorf("Update for orderbook failed: %v", err)
+	}
+}
+
+func TestGetUserInfo(t *testing.T) {
+	if l.APIKey == "" || l.APISecret == "" {
```

Comment:
Please change to areAPIKeysSet() bool function like other exchanges, then change all subsequent logic.